packages/edit-site/src/components/post-list/index.js

```diff
@@ -41,6 +41,22 @@ function useView( postType ) {
 	} = useLocation();
 	const history = useHistory();
 	const DEFAULT_VIEWS = useDefaultViews( { postType } );
+
+	/*
+	 * The selectedDefaultView represents one of the preconfigured views
+	 * the user can select from the sidebar.
+	 *
+	 * These views are defined by the following URL parameters:
+	 *
+	 * - activeView: the identifier. A number for custom views, a string for the default/bundled ones.
+	 * - isCustom: whether the view is a custom/user-defined or not.
+	 * - postType: the post type the view is for.
+	 *
+	 * Every time one of these parameters change, we reset the view completely.
+	 *
+	 * That's why this memoitization doesn't care about any other parameter (layout).
+	 * See https://github.com/WordPress/gutenberg/pull/63889
+	 */
```

Comment:
This probably triggers an eslint warning of a missing dependency. I think if we follow what I suggest here https://github.com/WordPress/gutenberg/pull/63854#discussion_r1688187227 there's a way to rework this code to be clearer.

Basically:

 - we don't need a memoized `selectedDefaultView` 

we need something like:

```
const [ view, setView ] = useState( { type: layout, ...getDefaultView( isCustom, activeView, postType ) } );

useEffect( () => {
   setView( ( prevView ) => ( { ...prevView, type: layout } ) );
}, [ layout ] );

useEffect(() => {
   setView( prev => ({ ...prev, ...getDefaultView( isCustom, activeView, postType ) });
}, [ isCustom, activeView, postType ] );
```

where `getDefaultView` returns the default view but without any "type" property. I think that code speaks for itself and clarifies the intent of any effect...

Reply:
I think this logic may work well, the activeView and layout may change at the same time but I think when an effect runs it changes the view directly, and on the setView, the prev already reflects the changes of the previous effect.